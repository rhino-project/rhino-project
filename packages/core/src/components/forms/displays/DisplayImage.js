import { useMemo } from 'react';
import { useController } from 'react-hook-form';

const DisplayImage = ({ accessor, altText, empty = '-', ...props }) => {
  const { path } = props;
  const {
    field: { value: fieldValue }
  } = useController({ name: path });

  const value = useMemo(() => (accessor ? accessor(fieldValue) : fieldValue), [
    accessor,
    fieldValue
  ]);

  if (!value) return <div>{empty}</div>;

  return (
    <img
      src={value}
      alt={altText || value}
      style={{ maxWidth: '100%', maxHeight: '100%' }}
    />
  );
};

export default DisplayImage;
