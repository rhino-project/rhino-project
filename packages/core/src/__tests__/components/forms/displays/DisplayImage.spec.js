import { sharedDisplayTests } from './sharedDisplayTests';
import DisplayImage from 'rhino/components/forms/displays/DisplayImage';

describe('DisplayImage', () => {
  sharedDisplayTests(DisplayImage);
});
