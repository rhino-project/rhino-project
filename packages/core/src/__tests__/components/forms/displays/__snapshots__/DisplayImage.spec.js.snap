// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayImage > renders inside of model context 1`] = `
<DocumentFragment>
  <div>
    -
  </div>
</DocumentFragment>
`;

exports[`DisplayImage > should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    -
  </div>
</DocumentFragment>
`;

exports[`DisplayImage > should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    baz
  </div>
</DocumentFragment>
`;

exports[`DisplayImage > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
